openbas-api/src/main/java/io/openbas/rest/exercise/ExerciseApi.java

```diff
@@ -324,14 +324,17 @@ public Iterable<Team> getExerciseTeams(@PathVariable String exerciseId) {
   @Transactional(rollbackOn = Exception.class)
   @PutMapping("/api/exercises/{exerciseId}/teams/add")
   @PreAuthorize("isExercisePlanner(#exerciseId)")
-  public Iterable<Team> addExerciseTeams(@PathVariable String exerciseId,
+  public Iterable<Team> addExerciseTeams(
+      @PathVariable String exerciseId,
       @Valid @RequestBody ExerciseUpdateTeamsInput input) {
     Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow();
     List<Team> teams = exercise.getTeams();
-    teams.addAll(fromIterable(teamRepository.findAllById(input.getTeamIds())));
+    List<Team> teamsToAdd = fromIterable(teamRepository.findAllById(input.getTeamIds()));
+    List<String> existingTeamIds = teams.stream().map(Team::getId).toList();
+    teams.addAll(teamsToAdd.stream().filter(t -> !existingTeamIds.contains(t.getId())).toList());
     exercise.setTeams(teams);
```

Comment:
Maybe we should use sets over lists more frequently?

Reply:
Good point to keep in mind